tracee-ebpf/tracee/events.go

```diff
@@ -49,35 +49,27 @@ func (t *Tracee) processEvents(done <-chan struct{}) error {
 			continue
 		}
 
-		args := make(map[string]interface{}, ctx.Argnum)
-		argMetas := make([]external.ArgMeta, ctx.Argnum)
-
 		eventDefinition, ok := EventsDefinitions[ctx.EventID]
 		if !ok {
 			t.handleError(fmt.Errorf("failed to get configuration of event %d", ctx.EventID))
 			continue
 		}
-		params := eventDefinition.Params
+
+		args := make([]external.Argument, 0, ctx.Argnum)
```

Comment:
Sorry if this is a dumb question.. but why can't we have a single type `Argument` with "Name, Type & Value" instead of having Argument with "ArgMeta & Value" ?

Reply:
Nevermind, I just saw that EventDefinition needs ArgMeta to be separated from the "Argument" struct.